app/src/main/java/xyz/whereuat/whereuat/LoginActivity.java

```diff
@@ -185,12 +197,74 @@ public void createNewAccount(View v) {
      * @return returns true if the phone number is a valid form
      */
     private boolean isValidPhoneForm(String phone_number) {
-        String pattern = "\\+1\\d{10}";
+        String pattern = "\\d{6}-\\d{4}";
         Pattern r = Pattern.compile(pattern);
         Matcher m = r.matcher(phone_number);
         return m.find();
     }
 
+    private void setAreaEditListeners() {
+        mAreaEdit.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                if (s.length() > AREA_CODE_LENGTH) {
+                    String overflow = s.subSequence(AREA_CODE_LENGTH, s.length()).toString();
+                    s.delete(AREA_CODE_LENGTH, s.length());
+                    mLineEdit.setText(overflow + mLineEdit.getText());
+                    mLineEdit.requestFocus();
+                    mLineEdit.setSelection(overflow.length());
+                }
+            }
+        });
+    }
+
+    private void setLineEditListeners() {
```

Comment:
Can you add some comments to each if case so it's really easy to know what each thing is supposed to do, rather than having to spend time logic-ing through it? Also, function docs? :)